plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java

```diff
@@ -374,9 +374,8 @@ public void reload(RepositoryMetadata newRepositoryMetadata) {
         readRepositoryMetadata();
 
         // Reload configs for S3RepositoryPlugin
-        final Map<String, S3ClientSettings> clientsSettings = S3ClientSettings.load(metadata.settings(), pluginConfigPath);
-        service.refreshAndClearCache(clientsSettings);
-        s3AsyncService.refreshAndClearCache(clientsSettings);
+        service.settings(metadata);
+        s3AsyncService.settings(metadata);
```

Comment:
Created backlog item for ITs for this : https://github.com/opensearch-project/OpenSearch/issues/10454 .